Start spells/talents/skills

Phatcat opened the pull request:
I've done a little something something regarding the CharStart...s , doesn't seem ready for merge yet but I'd love your input on some of it before moving forward, particularly the 'Start skills' and 'Remove CharStartTalents' commits.

Chaosvex commented on src/character/CharacterHandler.cpp:
```diff
@@ -175,13 +175,33 @@ void CharacterHandler::do_create(std::uint32_t account_id, std::uint32_t realm_i
 	} else { // could be intentional, so we'll keep going
-		LOG_DEBUG(logger_) << "No starting item data found for race" <<
+		LOG_DEBUG(logger_) << "No starting item data found for race " <<
 			character.race << ", class " << character.class_ << LOG_ASYNC;
```
Existing code but it'd be good to print the class/race as string inside. You can find other examples of that in this file (name.en_gb etc).

Phatcat replied:
I thought it was already doing that with the << character.race / << character.class_ - I will look into some of the other examples and alter it accordingly.

Phatcat replied:
Looking at the name.en_gb code I'm pretty sure that's what's already going on here as well.

Chaosvex commented on src/character/CharacterHandler.cpp:
```diff
@@ -392,7 +412,7 @@ bool CharacterHandler::validate_options(const Character& character, std::uint32_
 	}
 
-	// facial features (horns, markings, tusks, piercings, hair) validation
+	// facial features (horns, markings, tusks, piercings, beard) validation
```
This wasn't a mistake, please undo. ;)
```suggestion
	// facial features (horns, markings, tusks, piercings, hair) validation
```

Phatcat replied:
I thought it was since it was nested inside 'facial features', I'll undo this change again.

Chaosvex commented on src/character/CharacterHandler.cpp:
```diff
@@ -175,13 +175,33 @@ void CharacterHandler::do_create(std::uint32_t account_id, std::uint32_t realm_i
 
-	// populate spells information
+	// populate starting spells
+	auto spells = std::find_if(dbc_.char_start_spells.begin(), dbc_.char_start_spells.end(), [%](auto& record ) {
```
You want [&] to capture by reference, not [%].

Phatcat replied:
Ah, that's a typo.

Chaosvex replied:
Forgot to mention const on this one. Make *everything* const unless you have an actual need to modify it.

Phatcat replied:
Okay, seems reasonable.

Chaosvex commented on src/character/CharacterHandler.cpp:
```diff
@@ -175,13 +175,33 @@ void CharacterHandler::do_create(std::uint32_t account_id, std::uint32_t realm_i
 
-	// populate talents information
+	// populate starting skills
+	auto talents = std::find_if(dbc_.char_start_skills.begin(), dbc_.char_start_skills.end(), [%](auto& record ) {
```
```suggestion
	const auto& talents = std::find_if(dbc_.char_start_skills.begin(), dbc_.char_start_skills.end(), [&](auto& record ) {
```

Chaosvex commented on src/character/CharacterHandler.cpp:
```diff
@@ -175,13 +175,33 @@ void CharacterHandler::do_create(std::uint32_t account_id, std::uint32_t realm_i
 
-	// populate spells information
+	// populate starting spells
+	auto spells = std::find_if(dbc_.char_start_spells.begin(), dbc_.char_start_spells.end(), [%](auto& record ) {
```
```suggestion
	const auto& spells = std::find_if(dbc_.char_start_spells.begin(), dbc_.char_start_spells.end(), [&](auto& record ) {
```

Chaosvex commented on sql/mysql/full.sql:
```diff
@@ -78,6 +78,7 @@ CREATE TABLE `characters` (
   `x` float NOT NULL,
   `y` float NOT NULL,
   `z` float NOT NULL,
+  `o` float NOT NULL,
```
Wrong patch?

Phatcat replied:
Well, kind of, I didn't really know what to do with the change so I just committed it as it's own commit, I can get rid of it and place it in it's own PR.

Chaosvex replied:
Would be ideal.

natrist replied:
Why isn't o called facing here? I know it's for orientation, but using the internal names is a plus when reversing.

Phatcat replied:
because facing isn't the internal name as far as I can tell, neither is x, y or z; 
![image](https://user-images.githubusercontent.com/1907820/59564084-94312200-9042-11e9-9f78-0e4153d3ed19.png)

Phatcat replied:
(I am guessing POSSTAT0 is map, 1, 2, and 3 is x, y and z and 4 is orientation)

Phatcat replied:
Unless you are referring to the internal GO names, in which case x, y and z should be renamed pos_x, pos_y and pos_z as well;
![image](https://user-images.githubusercontent.com/1907820/59564148-ab244400-9043-11e9-8917-6291bb19f7b4.png)

Chaosvex commented on src/character/CharacterHandler.cpp:
```diff
@@ -175,13 +175,33 @@ void CharacterHandler::do_create(std::uint32_t account_id, std::uint32_t realm_i
+		return record.second.race_id == character.race && record.second.class__id == character.class_;
+	});
+
+	if (spells != dbc_.char_start_spells.end()) {
```
Slight formatting change.
```suggestion
	if(spells != dbc_.char_start_spells.end()) {
```

Phatcat committed:
Cleanup CharStartBase.xml

If this is unutitilized in the project shouldn't it just be outright removed?

Phatcat committed:
Replace CharStartTalents with CharStartSkills

Characters don't start with any talents at all (they're not even unlocked until level 10).
Not even the death knight in wotlk starts with any talents.

CharStartSkills should perhaps not have SkillRaceClassInfo as it's parent in the 'skill' field but I couldn't find anything more applicable, and we need to get the skills from -somewhere-

Phatcat committed:
Expand on initial start spells impl

Expand slightly upon the starting spell implementation
